pynetdicom/dul.py

```diff
@@ -453,9 +453,9 @@ def _send(self, pdu: _PDUType) -> None:
         pdu : pynetdicom.pdu.PDU
             The PDU to be encoded and sent to the peer.
         """
-        sock = cast("AssociationSocket", self.socket)
-        sock.send(pdu.encode())
-        evt.trigger(self.assoc, evt.EVT_PDU_SENT, {"pdu": pdu})
+        if self.socket is not None:
```

Comment:
what if self.socket.socket is None? shouldn't there be an if statement for that just in case? i think it might be the solution for this issue https://github.com/pydicom/pynetdicom/issues/806

Reply:
Ideally the state machine and reactors should manage things so it never tries to send data while None